### PR TITLE
Add default port when choose database

### PR DESCRIPTION
**Pull request by krolow**
This feature auto fill the port when choose one database...

We can not merge it yet as it depends of generate a tag for https://github.com/sqlectron/sqlectron-core/pull/12

I'll just leave it open to review code and once we generate a tag for  sqlectron-core, i can bump the https://github.com/sqlectron/sqlectron-gui/blob/master/package.json#L51 dependency

### CODE REVIEW

File: src/renderer/components/server-modal-form.jsx [hunk 1]
```diff
@@ -130,6 +131,15 @@ export default class ServerModalForm extends Component {
     return hasError ? 'error' : '';
   }
 
+  handleOnClientChange(client) {
+    this.setState({ client });
+
+    const clientConfig = CLIENTS.find(entry => entry.value === client);
+    if (clientConfig && clientConfig.defaultPort) {
```
Comment (maxcnunes) [inline src/renderer/components/server-modal-form.jsx]: Shouldn't it check if the port field has a value already?

Comment (krolow) [inline src/renderer/components/server-modal-form.jsx]: i thought about that... i have even added that... but imagine that you are editing one connection you have added mysql but actually it was pgsql... should we disallow default port?

for me it makes sense too to not change when have state but while i was testing i found it a bit weird...

File: src/renderer/components/server-modal-form.jsx [hunk 2]
```diff
@@ -232,7 +242,7 @@ export default class ServerModalForm extends Component {
                     name="port"
                     maxLength="5"
                     placeholder="Port"
-                    value={this.state.port}
+                    value={this.state.port || this.state.defaultPort}
```
Comment (krolow) [inline src/renderer/components/server-modal-form.jsx]: @maxcnunes so i have change here so only if user has not choose any port we gonna apply default one

**Comment by colecmc** [comment]
SQLectron is an awesome product. Thanks for sharing it.

I am using Version 1.21.0 (1.21.0.451) on a Mac. And I cannot connect to a Microsoft SQL DB because a default port is being added.
When I remove the default port, I am able to test the connection successfully, however I'm not able to save it.

What am I doing wrong? Please help.

**Comment by eMerzh** [comment]
@colecmc could you open an separate issue about that ?  and not sure it will change but updating to the latest version could help too ;)

Thanks

**Comment by colecmc** [comment]
OK, I've opened [334](https://github.com/sqlectron/sqlectron-gui/issues/334)

**Pull request merged**